bedrock/src/smart_account/nonce.rs

```diff
@@ -18,6 +18,12 @@ use crate::primitives::BEDROCK_NONCE_PREFIX_CONST;
 pub enum TransactionTypeId {
     /// ERC-20 transfer
     Transfer = 1,
+    /// `WorldGiftManager` gift
+    WorldGiftManagerGift = 2,
```

Comment:
given this is potentially short lived, or at least not standard transactions, can we please use higher types and reserve these ones (e.g. 127, 128, 129)